skops/cli/_update.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import zipfile
+from pathlib import Path
+
+from skops.cli._utils import get_log_level
+from skops.io import dump, load
+from skops.io._protocol import PROTOCOL
+
+
+def _update_file(
+    input_file: str | Path,
+    output_file: str | Path | None = None,
+    inplace: bool = False,
+    logger: logging.Logger = logging.getLogger(),
+) -> None:
+    """Function that is called by ``skops update`` entrypoint.
+
+    Loads a skops model from the input path, converts to the current skops format, and
+    saves to output file.
+
+    Parameters
+    ----------
+    input_file : str, or Path
+        Path of input skops model to load.
+
+    output_file : str, or Path, default=None
+        Path to save the updated skops model to.
+
+    inplace : bool, default=False
+        Whether to update and overwrite the input file in place.
+
+    logger : logging.Logger, default=logging.getLogger()
+        Logger to use for logging.
+    """
+    if inplace:
+        if output_file is None:
+            output_file = input_file
+        else:
+            raise ValueError(
+                "Cannot specify both an output file path and the inplace flag. Please"
+                " choose whether you want to create a new file or overwrite the input"
+                " file."
+            )
+
```

Comment:
Basically, for the case that `output_file is None and not inplace`, further below, you print a message that the file can be updated if the protocol is different. So basically you added a new functionality, which is to check if a file needs an update. This is fine for me, I see how it can be useful, but it should be documented, otherwise I think users will not discover it ("To check if a file needs to be updated, ...").

I would also prefer to reorder this function a bit, with all argument checks coming at the top and then using early returns to get rid of some of the nesting.

So basically:

1. Check if both inplace and output file (so the same as right now)
2. If protocol is identical, log message and early return
3. Check if neither inplace, nor output file, check protocol, log message, early return

WDYT?

Reply:
You are right, I was also not a fan of the if-else nesting, thank you for the tip. Let me know if this is what you had in mind

Reply:
You are right, I was also not a fan of the if-else nesting, thank you for the tip. Let me know if this is what you had in mind